pkg/e2/v1beta1/options.go

```diff
@@ -183,3 +209,10 @@ func WithE2TPort(port int) Option {
 		options.Service.Port = port
 	})
 }
+
+// WithSubscriptionTransactionTimeout sets a timeout value for subscriptions
+func WithSubscriptionTransactionTimeout(transactionTimeout time.Duration) SubscribeOption {
```

Comment:
I think this should just be `WithTransactionTimeout`. It will only compile as an argument to a `Subscribe` call, so Subscription is redundant.